### PR TITLE
Add comments, break down certificate parsing

### PR DESCRIPTION
**Pull request by tnarg999**

### CODE REVIEW

File: linter/lintfuncs.go [hunk 1]
```diff
@@ -11,24 +11,28 @@ const (
...
+// LintProducedAtDate checks that an OCSP Response ProducedAt date is no more than ProducedAtLimit in the past
+// Source: Apple Lint 03
 func LintProducedAtDate(resp *ocsp.Response) error {
 	limit, err := time.ParseDuration(ProducedAtLimit)
 	if err != nil {
 		return err
 	}
 	if time.Since(resp.ProducedAt) > limit {
-		return fmt.Errorf("OCSP Response producedAt date is more than %s in the past", ProducedAtLimit)
+		return fmt.Errorf("OCSP Response producedAt date, %s, is more than %s in the past", resp.ProducedAt, ProducedAtLimit)
```
Comment (aokeke) [inline linter/lintfuncs.go]: Really minor, but the use of commas here seem a bit weird. I might use single quotes.

`"OCSP Response producedAt date '%s' is more than %s in the past"`. In python I'd use %r but I don't know if there is an equivalent in Golang.

File: linter/lintfuncs.go [hunk 2]
```diff
@@ -11,24 +11,28 @@ const (
...
+// LintThisUpdateDate checks that an OCSP Response ThisUpdate date is no more than ThisUpdateLimit in the past
+// Source: Apple Lint 03
 func LintThisUpdateDate(resp *ocsp.Response) error {
 	limit, err := time.ParseDuration(ThisUpdateLimit)
 	if err != nil {
 		return err
 	}
 	if time.Since(resp.ThisUpdate) > limit {
-		return fmt.Errorf("OCSP Response thisUpdate date is more than %s in the past", ThisUpdateLimit)
+		return fmt.Errorf("OCSP Response thisUpdate date, %s, is more than %s in the past", resp.ThisUpdate, ThisUpdateLimit)
```
Comment (aokeke) [inline linter/lintfuncs.go]: Same comment here

File: ocsp_status.go [hunk 3]
```diff
@@ -38,35 +27,62 @@ func checkFromFile(respFile string) error {
 	return nil
 }
 
+// checkFromCert takes a path to an ASN.1 DER encoded certificate file and constructs and sends an OCSP request
```
Comment (aokeke) [inline ocsp_status.go]: This line looks a bit long (111 characters). Normally we limit things to 80 characters. Can you move some onto the next line.

File: ocsp_status.go [hunk 4]
```diff
@@ -38,35 +27,62 @@ func checkFromFile(respFile string) error {
...
+		return fmt.Errorf("Error getting OCSP Response: %w", err)
 	}
 
 	return ocsptools.ParseAndLint(ocspResp, issuerCert)
-
 }
 
-func checkFromURL(serverURL string, shouldPrint bool, isPost bool, ocspURL string, dir string, hash crypto.Hash) error {
-	tlsConn, err := createConn(serverURL)
+// checkFromURL takes a server URL and constructs and sends an OCSP request to check that URL's certificate
```
Comment (aokeke) [inline ocsp_status.go]: line length again

File: ocsp_status.go [hunk 5]
```diff
@@ -118,40 +131,42 @@ func main() {
...
+	noStaple := flag.Bool("nostaple", false, "Whether to send an OCSP request regardless of if there is a stapled OCSP response")	
 
 	flag.Parse()
 
 	if *inresp && *incert {
 		panic("This tool can only parse one file format at a time. Please use only one of -inresp or -incert.")
 	}
 
-	if *inresp {
+	if *inresp { // reading in OCSP response files
```
Comment (aokeke) [inline ocsp_status.go]: The comments in line seems a bit weird IMO, can they be on their own line?

File: ocsptools/ocsp_tools.go [hunk 6]
```diff
@@ -58,13 +75,34 @@ func GetCertFromIssuerURL(issuerURL string) (*x509.Certificate, error) {
...
 // leafCert is the root certificate (first certificate in the chain)
-// issuerCert is the last certificate in the chain
-// reqMethod is either GET or POST (TODO: change reqMethod to not be string)
+// issuerCert is the certificate of the issuer of the leafCert
+// reqMethod is either GET or POST
+// hash is the hash to use to encode the request (either SHA1 or SHA256 right now)
 func CreateOCSPReq(ocspURL string, leafCert *x509.Certificate, issuerCert *x509.Certificate, reqMethod string, hash crypto.Hash) (*http.Request, error) {
 	if ocspURL == "" {
-		ocspURL = leafCert.OCSPServer[0]
+		// leafCert probably is an intermediary, which may not be required to have an OCSP responder
```
Comment (aokeke) [inline ocsptools/ocsp_tools.go]: This also goes beyond the 80 character line length, can it be broken into a 2-line comment.

File: test/lint_test.go [hunk 7]
```diff
@@ -13,7 +13,9 @@ const (
 	TestRespDates = "./test-resps/oldfbresp"
 )
 
-
+// TestLintProducedAtDate tests linter.LintProducedAtDate, which checks that an OCSP Response ProducedAt date
```
Comment (aokeke) [inline test/lint_test.go]: comment line length again.

File: test/lint_test.go [hunk 8]
```diff
@@ -40,6 +42,9 @@ func TestLintProducedAtDate(t *testing.T) {
     }
 }
 
+// TestLintThisUpdateDate tests linter.LintThisUpdateDate, which checks that an OCSP Response ThisUpdate date
```
Comment (aokeke) [inline test/lint_test.go]: here also

**Review by aokeke — CHANGES_REQUESTED** [review]
Just some small formatting requests.

**Comment by tnarg999** [comment]
So my Chrome Remote Desktop, which I have been using to work, is unbearably slow now. I've realized that since this is an open source project I should be able to just work on my local Mac, which is what I'm doing now. Unfortunately I made a commit here without first resetting my commit email address to be my google email, and the CLA bot is mad at me.

I've created a new PR that resolves this issue so I'll close this one.

**Pull request closed without merging**